### ISSUE TITLE
N814 false positive: `from decimal import Decimal as D`

### ISSUE THREAD

**Issue by rdaysky** [astral-sh/ruff#13543]
N814 ([camelcase-imported-as-constant](https://docs.astral.sh/ruff/rules/camelcase-imported-as-constant/)) triggers even if the alias is a single uppercase letter in [code like](https://docs.python.org/dev/library/statistics.html#statistics.mean)
```
from decimal import Decimal as D
from fractions import Fraction as F
```
but perhaps it shouldn’t?

**Comment by charliermarsh** [comment]
I tend to agree that we shouldn't flag this.

**Comment by rdaysky** [comment]
It’s also an interesting question whether `from collections.abc import AsyncGenerator as AG` is acceptable style for a module that defines a lot of them.

**Comment by dhruvmanila** [comment]
Did you mean `N817` instead of `N814`? These won't be flagged by the latter as it'll be considered as an acronym for the imported name. See https://play.ruff.rs/9326f4aa-a8bf-47c0-a45f-5b3d04b902c7.

**Comment by dhruvmanila** [comment]
You can also use the [`extend-aliases`](https://docs.astral.sh/ruff/settings/#lint_flake8-import-conventions_extend-aliases) option here:

```toml
[tool.ruff.lint.flake8-import-conventions.extend-aliases]
"collections.abc.AsyncGenerator" = "AG"
"decimal.Decimal" = "D"
"fractions.Fraction" = "F"
```

**Comment by dhruvmanila** [comment]
I think I'd suggest to use the configuration instead of adding special casing for certain imports. What do you think? @rdaysky
